e2e/cypress/integration/notifications/desktop_notifications_spec.js

```diff
@@ -56,6 +56,37 @@ describe('Desktop notifications', () => {
         });
     });
 
+    it('MM-T482 Desktop Notifications - (at) here not rec\'d when logged off', () => {
+        cy.apiCreateUser({}).then(({user}) => {
+            cy.apiAddUserToTeam(testTeam.id, user.id);
+            cy.apiLogin(user);
+
+            // # Ensure notifications are set up to fire a desktop notification if are mentioned.
+            changeDesktopNotificationSettingsAs('#desktopNotificationMentions');
+            stubNotificationAs('withoutNotification', 'granted');
+
+            cy.apiGetChannelByName(testTeam.name, 'Off-Topic').then((res) => {
+                const channel = res.body;
+
+                // # Logout the user.
+                cy.apiLogout();
+
+                // Have another user send a post.
+                cy.postMessageAs({sender: testUser, message: '@here', channelId: channel.id});
+            });
+
+            // # Login with the user.
+            cy.apiLogin(user).then(() => {
+                // Visit town-square.
+                cy.visit(`/${testTeam.name}/channels/town-square`);
+
+                // * Desktop notification is not received
+                cy.wait(TIMEOUTS.HALF_SEC);
+                cy.get('@withoutNotification').should('not.have.been.called');
```

Comment:
The verification seemed to be lacking some checks on channels' mention indicator, highlight and marked as unread. As well as verification whether no email notification is received. For email notification, you'll be able to access email inbox via local Inbucket API. Look for `getRecentEmail` to see how it works.

Reply:
I add the expectations from the test case:
1- Channel mention indicator's absence and channel has unreads.
2- Highlighted `@here` mention.
3- No email notification is received: did this with verifying only joining to the server e-mail exist in the inbucket.

Reply:
Thank you so much! Sounds good to me.